### PR TITLE
ENH: Remove journal items when banning user

### LINKED ISSUE #741: ENH: Remove journal item when deleting post
Follow-on to #728 and #730. Forum posts are logged to DNN core journal. When banning a user, posts are deleted but also should be removed from journal.

### PR DESCRIPTION
**Pull request by johnhenley**
### Description of PR...
Removes journal items when deleting posts / banning user

## How did you test these updates?  

Local install

Journal items before:
image

after:
image

Notification to Moderator(s) now includes items removed:
image

image

## PR Template Checklist

- [ ] Fixes Bug
- [X] Feature solution
- [ ] Other
- [ ] Requires documentation updates  
- [ ] I've updated the documentation already  

## Please mark which issue is solved

Close #741

### CODE REVIEW

File: Dnn.CommunityForums/Controllers/UserController.cs [hunk 1]
```diff
@@ -18,16 +18,53 @@
...
+        {
+            internal int ForumId { get; set; }
+            internal int TopicId { get; set; }
+            internal int ReplyId { get; set; }
+        }
         internal static void BanUser(int PortalId, int ModuleId, int UserId)
         {
             if (UserId > -1)
             {
+                string sql = "SELECT ft.ForumId, ft.TopicId, r.ReplyId " +
```
Comment (WillStrohl) [inline Dnn.CommunityForums/Controllers/UserController.cs:39]: @johnhenley Shouldn't we put this in as a SProc? 🤔

**Review by WillStrohl — COMMENTED** [review]
Just added a question.

**Comment by WillStrohl** [comment]
Oh, and there's a conflict in the SqlDataProvider.

**Comment by johnhenley** [comment]
> Oh, and there's a conflict in the SqlDataProvider.

Naturally. 
I moved to stored procedure, and also put the items that are removed into the moderator notification.

**Pull request merged**